src/Symfony/Component/Process/Process.php

```diff
@@ -1671,13 +1671,8 @@ private function replacePlaceholders(string $commandline, array $env)
 
     private function getDefaultEnv(): array
     {
-        $env = [];
-
-        foreach ($_SERVER as $k => $v) {
-            if (\is_string($v) && false !== $v = getenv($k)) {
-                $env[$k] = $v;
-            }
-        }
+        $env = getenv();
+        $env = array_intersect_key($env, $_SERVER) ?: $env;
```

Comment:
This means if the current running process has added env vars putenv(), they still won't be detected though. One has to update $_SERVER + do putenv() for it to be taken into account.

Reply:
That's right, we are intentionally not compatible with putenv() here, because using putenv leads to blank pages in Windows